src/metadataGeneration/typeResolver.ts

```diff
@@ -254,6 +254,29 @@ export class TypeResolver {
         return stringMetaType;
       }
 
+      const type = this.current.typeChecker.getTypeFromTypeNode(this.typeNode);
+      const toJSON = this.current.typeChecker.getPropertyOfType(type, 'toJSON');
+      if (toJSON) {
+        const declaration = toJSON.declarations[0] as ts.MethodDeclaration;
+
+        let nodeType = declaration.type;
+        if (!nodeType) {
+          const typeChecker = this.current.typeChecker;
+          const signature = typeChecker.getSignatureFromDeclaration(declaration);
```

Comment:
We should not use the typeChecker by default. Best way is always to recursively call the TR on the typeNode and only fall back to the inferred type if there's no explicit typeNode.

Reply:
I don't understand, what do you mean by `TR`? And you're saying we're using the typeChecker by default but we only do it if `declaration.type` is undefined

Reply:
You're right, sorry about that, I missed that.

Reply:
So can I resolve this?